### PR TITLE
Feature/fix cfd login

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x]  userContext
- [x] DepostiModal
- [x] OpenSubTab
- [x] OpenSubTabMobile
- [x] TradeTab
- [x] TradeTabMobile
- [x] PositionClosedModal
- [x] PositionOpenModal
- [x] SignatureProcessModal
- [x] UpdateFormModal
- [x] useShareProcess

### CHECK LIST
- [x]  #722 
- [x] #702 
- [x] #701 
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component:  0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/lib/common.ts [hunk 1]
```diff
@@ -550,3 +554,13 @@ export const findCodeByReason = (reason: string): ICode | undefined => {
 
   return ReasonToCode[reason] as ICode;
 };
+
+export const validateCFD = (fee: number, margin: number, amount: number) => {
+  if (fee > margin * MAX_FEE_RATE || fee < MIN_FEE_RATE) {
```
Comment (arealclimber) [inline src/lib/common.ts:559]: fee > value * MAX_FEE_RATE

Comment (arealclimber) [reply src/lib/common.ts:559]: 只要檢查 fee 百分比

Comment (arealclimber) [reply src/lib/common.ts:559]: fee, value, margin

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**